src/main/java/mate/dao/DriverDaoImpl.java

```diff
@@ -104,9 +108,30 @@ public boolean delete(Long id) {
     private Driver getDriver(ResultSet resultSet) throws SQLException {
         Long id = resultSet.getObject("id", Long.class);
         String name = resultSet.getString("name");
+        String login = resultSet.getString("login");
+        String password = resultSet.getString("password");
         String licenseNumber = resultSet.getString("license_number");
         Driver driver = new Driver(name, licenseNumber);
+        driver.setLogin(login);
+        driver.setPassword(password);
         driver.setId(id);
         return driver;
     }
+
+    @Override
+    public Optional<Driver> findByLogin(String login) {
+        String query = "SELECT * FROM drivers WHERE login = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setString(1, login);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            Driver driver = null;
```

Comment:
Let's declare Driver driver = null; before try-block.

Reply:
I think that we should create object as close as possible to its first use

Reply:
I leave that to the mentors